validation/validateUserForm.ts

```diff
@@ -0,0 +1,42 @@
+import { IValidation } from "../models/validationModel";
+
+export const validateForm = <IValidation>(value: string, id: string) => {
+  if (id === "firstName") {
```

Comment:
A simpler construction instead of having a large if else for all different types of inputs it to send in rules to each input component that are checked, much like how React hook form does it. This solution would soon get very large if there are many type of ids to check against.